frontend/src/store/credential.js

```diff
@@ -73,91 +83,96 @@ export const useCredentialStore = defineStore('credential', () => {
     return Object.values(state.quotas)
   })
 
   async function createCredential (params) {
-    const { data: { secretBinding, secret } } = await api.createCloudProviderCredential({ secretBinding: params.secretBinding, secret: params.secret })
-    _updateCloudProviderCredential({ secretBinding, secret })
-    appStore.setSuccess(`Cloud Provider credential ${secretBinding.metadata.name} created`)
+    const { data: { binding, secret } } = await api.createCloudProviderCredential({ binding: params.binding, secret: params.secret })
+    _updateCloudProviderCredential({ binding, secret })
+    appStore.setSuccess(`Cloud Provider credential ${binding.metadata.name} created`)
   }
 
   async function updateCredential (params) {
-    const { data: { secretBinding, secret } } = await api.updateCloudProviderCredential({ secretBinding: params.secretBinding, secret: params.secret })
-    _updateCloudProviderCredential({ secretBinding, secret })
-    appStore.setSuccess(`Cloud Provider credential ${secretBinding.metadata.name} updated`)
+    const { data: { secret } } = await api.updateCloudProviderCredential({ secret: params.secret })
+    _updateCloudProviderCredential({ secret })
+    appStore.setSuccess(`Cloud Provider credential ${params.binding.metadata.name} updated`)
   }
 
-  async function deleteCredential (name) {
-    const namespace = authzStore.namespace
-
-    await api.deleteCloudProviderCredential({ namespace, name })
+  async function deleteCredential ({ bindingKind, bindingNamespace, bindingName }) {
+    await api.deleteCloudProviderCredential({ bindingKind, bindingNamespace, bindingName })
     await fetchCredentials()
-    appStore.setSuccess(`Cloud Provider credential ${name} deleted`)
+    appStore.setSuccess(`Cloud Provider credential ${bindingName} deleted`)
   }
 
-  const infrastructureSecretBindingsList = computed(() => {
-    return filter(secretBindingList.value, secretBinding => {
-      return cloudProfileStore.sortedProviderTypeList.includes(secretBinding.provider?.type)
+  const infrastructureBindingList = computed(() => {
+    return filter(cloudProviderBindingList.value, binding => {
+      return isInfrastructureBinding(binding, cloudProfileStore)
     })
   })
 
-  const dnsSecretBindingsList = computed(() => {
-    return filter(secretBindingList.value, secretBinding => {
-      return gardenerExtensionStore.dnsProviderTypes.includes(secretBinding.provider?.type) && hasOwnSecret(secretBinding) // setting secret binding not supported
+  const dnsBindingList = computed(() => {
+    return filter(cloudProviderBindingList.value, binding => {
+      return isDnsBinding(binding, gardenerExtensionStore) &&
+        !isSharedCredential(binding)
     })
   })
 
+  const secretBindingList = computed(() => {
+    return filter(cloudProviderBindingList.value, isSecretBinding)
+  })
+
+  const credentialsBindingList = computed(() => {
+    return filter(cloudProviderBindingList.value, isCredentialsBinding)
```

Comment:
why not use 
`const secretBindings = Object.values(state.secretBindings)`
or
`credentialsBindings = Object.values(state.credentialsBindings)`
respectively

Reply:
yes now that we removed the decoration we can do that...